src/components/IdScanner/IdScanner.tsx

```diff
@@ -161,15 +161,18 @@ export const IdScanner: FunctionComponent<IdScanner> = ({
 
   const checkIfInInterestArea: BarCodeScannedCallback = (event) => {
     const bounds = event.bounds?.origin;
+    const width = event.bounds?.size.width;
+    const height = event.bounds?.size.height;
```

Comment:
Why was this extracted? Did you face issues?

Reply:
there were errors that `event.bounds` can be undefined, which causes error in the `ifs` because we're operating `numbers | undefined`.
thus the `bounds && ... ` and `width` `height`

Reply:
Not sure if we prefer this instead?

```
  const checkIfInInterestArea: BarCodeScannedCallback = (event) => {
    const { bounds } = event;

    if (bounds && interestArea) {
      const { origin, x: boundsX, y: boundsY } = bounds;
      const { x: interestAreaX, y: interestAreaY } = interestArea;

      if (origin && boundsX && boundsY && interestAreaX && interestAreaY) {
        const { size } = origin;

        if (size) {
          const { width, height } = size;

          if (
            boundsX >= interestAreaX &&
            boundsY >= interestAreaY &&
            boundsX + width <= interestAreaX + width &&
            boundsY + height <= interestAreaY + height
          ) {
            onBarCodeScanned(event);
          }
        }
      }
    }
  };
```

Reply:
discussed